apps/studio/src/server/modules/version/version.service.ts

```diff
@@ -0,0 +1,98 @@
+import type { SelectExpression } from "kysely"
+import { type DB } from "~prisma/generated/generatedTypes"
+
+import type { SafeKysely } from "../database"
+import { db } from "../database"
+import { getPageById, updatePageById } from "../resource/resource.service"
+
+const defaultVersionSelect: SelectExpression<DB, "Version">[] = [
+  "Version.id",
+  "Version.versionNum",
+  "Version.resourceId",
+  "Version.blobId",
+  "Version.publishedAt",
+]
+
+export const getVersionById = ({ versionId }: { versionId: string }) =>
+  db
+    .selectFrom("Version")
+    .where("Version.id", "=", versionId)
+    .select(defaultVersionSelect)
+    .executeTakeFirstOrThrow()
+
+const createVersion = async (
+  db: SafeKysely,
+  props: {
+    versionNum: number
+    resourceId: number
+    blobId: number
+    publisherId: string
+  },
+) => {
+  const { versionNum, resourceId, blobId, publisherId } = props
+  const addedVersion = await db
+    .insertInto("Version")
+    .values({
+      versionNum,
+      resourceId: String(resourceId),
+      blobId: String(blobId),
```

Comment:
not directly actionable but how will we utilise this `blobId`? the reason i'm asking is because this is a reference so even if we publish, the value of the blob itself can still change.

Reply:
Once published the blob shouldn't change, it will be tied closely to the version. on the page resource, draftBlobId is immediately set to null so any future edits will be captured as a new blob